--- a/extendr-api/build.rs
+++ b/extendr-api/build.rs
@@ -6,6 +6,7 @@
     println!("cargo:rustc-check-cfg=cfg(use_r_ge_version_14)");
     println!("cargo:rustc-check-cfg=cfg(use_r_ge_version_15)");
     println!("cargo:rustc-check-cfg=cfg(use_r_altlist)");
+    println!("cargo:rustc-check-cfg=cfg(r_4_5)");
 
     // The R version information is needed to handle the API differences
     // between versions. `These DEP_R_R_VERSION_*` are provided by extendr-ffi
@@ -40,4 +41,9 @@
     if &*major >= "4" && &*minor >= "4" {
         println!("cargo:rustc-cfg=use_objsxp");
     }
+
+    // R 4.5 introduced new API functions replacing non-API calls
+    if &*major >= "4" && &*minor >= "5" {
+        println!("cargo:rustc-cfg=r_4_5");
+    }
 }
