--- a/extendr-ffi/src/lib.rs
+++ b/extendr-ffi/src/lib.rs
@@ -320,6 +320,7 @@
     pub fn R_IsPackageEnv(rho: SEXP) -> Rboolean;
     pub fn R_MakeUnwindCont() -> SEXP;
     #[doc = "Missing argument marker"]
+    #[cfg(r_4_5)]
     pub static R_MissingArg: SEXP;
     pub fn R_NamespaceEnvSpec(rho: SEXP) -> SEXP;
     #[doc = "Registry for registered namespaces"]
@@ -350,8 +351,6 @@
     pub static R_Srcref: SEXP;
     pub fn R_tryEval(arg1: SEXP, arg2: SEXP, arg3: *mut ::std::os::raw::c_int) -> SEXP;
     pub fn R_tryEvalSilent(arg1: SEXP, arg2: SEXP, arg3: *mut ::std::os::raw::c_int) -> SEXP;
-    #[doc = "Unbound marker"]
-    pub static R_UnboundValue: SEXP;
     pub fn R_unif_index(arg1: f64) -> f64;
     pub fn R_Unserialize(ips: R_inpstream_t) -> SEXP;
     pub fn R_UnwindProtect(
